src/screens/Rari/RariTransfer.js

```diff
@@ -0,0 +1,352 @@
+import { Keyboard } from 'react-native';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import debounce from 'lodash.debounce';
+import isEmpty from 'lodash.isempty';
+import t from 'translations/translate';
+
+import { BaseText } from 'components/Typography';
+import FeeLabelToggle from 'components/FeeLabelToggle';
+import Toast from 'components/Toast';
+import ContactDetailsModal from 'components/ContactDetailsModal';
+import Modal from 'components/Modal';
+
+import SendContainer from 'containers/SendContainer';
+
+import { isEnoughBalanceForTransactionFee } from 'utils/assets';
+import { reportErrorLog, noop } from 'utils/common';
+import { getContactWithEnsName } from 'utils/contacts';
+import { isEnsName } from 'utils/validators';
+
+import { resetEstimateTransactionAction, estimateTransactionAction } from 'actions/transactionEstimateActions';
+import { addContactAction } from 'actions/contactsActions';
+
+
+import { ETH } from 'constants/assetsConstants';
+import { RARI_TRANSFER_REVIEW } from 'constants/navigationConstants';
+import { RARI_TOKENS_DATA } from 'constants/rariConstants';
+
+import { accountBalancesSelector } from 'selectors/balances';
+import { contactsSelector } from 'selectors';
+import { useGasTokenSelector } from 'selectors/smartWallet';
+
+import type { RootReducerState, Dispatch } from 'reducers/rootReducer';
+import type { NavigationScreenProp } from 'react-navigation';
+import type { TransactionFeeInfo } from 'models/Transaction';
+import type { Balances, AssetData } from 'models/Asset';
+import type { RariPool } from 'models/RariPool';
+import type { Contact } from 'models/Contact';
+import type { Option } from 'models/Selector';
+
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  feeInfo: ?TransactionFeeInfo,
+  isEstimating: boolean,
+  estimateErrorMessage: ?string,
+  userDepositInRariToken: {[RariPool]: number},
+  contacts: Contact[],
+  addContact: (contact: Contact) => void,
+  estimateTransaction: (recipient: string, value: number, assetData: AssetData) => void,
+  useGasToken: boolean,
+  balances: Balances,
+  resetEstimateTransaction: () => void,
+};
+
+const renderFeeToggle = (
+  txFeeInfo: ?TransactionFeeInfo,
+  showFee: boolean,
+  feeError: ?string,
+  isLoading: boolean,
+  enoughBalance: boolean,
+) => {
+  if (!showFee || !txFeeInfo) return null;
+
+  const { fee, gasToken } = txFeeInfo;
+
+  return (
+    <>
+      <FeeLabelToggle txFeeInWei={fee} gasToken={gasToken} isLoading={isLoading} hasError={!enoughBalance} />
+      {!!feeError && <BaseText style={{ marginTop: 15 }} center secondary>{feeError}</BaseText>}
+    </>
+  );
+};
+
+const RariTransferScreen = ({
+  navigation,
+  feeInfo,
+  isEstimating,
+  estimateErrorMessage,
+  userDepositInRariToken,
+  contacts,
+  addContact,
+  estimateTransaction,
+  useGasToken,
+  balances,
+  resetEstimateTransaction,
+}) => {
```

Comment:
I'd only like to add that it looks like prop types are missing here: 
```suggestion
}: Props) => {
```